Delete video

asebrech opened the pull request:
This pull request introduces new functionality to allow deleting individual movies or all movies, including their associated files, through the API. The main changes include adding new endpoints and implementing corresponding service methods to handle movie deletion and cleanup, with robust error handling and reporting.

**API Enhancements:**

* Added two new endpoints to the `torrent` route: `DELETE /torrent/:id` for deleting a specific movie by TMDB ID, and `DELETE /torrent/` for deleting all movies.
* Updated `TorrentController` to include new methods `delete` and `deleteAll`, which handle incoming delete requests, validate input, call the cleanup service, and format responses. Error handling and response formatting helpers were also added.
* Injected the new `MovieCleanupService` into `TorrentController` to support the delete operations. [[1]](diffhunk://#diff-11c34de6555fdd58823dc31133d94bf1b627d219deb73e70b70ecb2591576eabR3) [[2]](diffhunk://#diff-11c34de6555fdd58823dc31133d94bf1b627d219deb73e70b70ecb2591576eabL12-R14)

**Service Layer Additions:**

* Implemented `deleteMovie` and `deleteAllMovies` methods in `MovieCleanupService`, which handle deletion of movie database entries and associated files, track freed disk space, and collect error messages.
* Added a private helper `deleteSingleMovie` in `MovieCleanupService` to encapsulate the logic for deleting a single movie and its files, with validation and error reporting.

Commit:
delete by id

Commit:
delete all

Commit:
merge refacto

Commit:
better typing

Commit:
deduplicat format

Commit:
sanitaze id

LaBoiteDeCarton commented on apps/back/start/routes.ts:
```diff
@@ -49,6 +49,8 @@ router
     router.get('/:id', [TorrentController, 'torrent'])
     router.get('/:resolution/:id', [TorrentController, 'ready'])
+    router.delete('/', [TorrentController, 'deleteAll'])
+    router.delete('/:id', [TorrentController, 'delete'])
```
N'importe quel utilisateur connecté peux supprimer les film du serveur. Faudrait un middleware specifique a un utilisateur type admin. (c'est chiant faut modifier la db)

asebrech replied:
Bien vu, j'avais prévu de faire ça puis ça m'est sorti de la tête, merci !

Commit:
admin middleware

LaBoiteDeCarton committed:
Merge branch 'staging' into delete-video

LaBoiteDeCarton commented on apps/back/app/controllers/torrent_controller.ts:
```diff
@@ -40,4 +59,89 @@ export default class TorrentController {
+        })
+      }
+
+      const result = await this.movieCleanupService.deleteMovie(tmdbId)
```
<img width="939" height="256" alt="Capture d’écran 2025-08-18 à 22 03 44" src="https://github.com/user-attachments/assets/9540a695-9bcf-47f3-9f31-f5355054cb5a" />

Si le movie est en cours de conversion l'app crash au moment de delete

Commit:
fix: prevent deletion of movies during conversion/download

- Add checks in MovieCleanupService to prevent deletion of movies that are currently being converted or downloaded
- Return appropriate error messages when attempting to delete processing movies
- Update deleteMovie() to check conversion and download status
- Update deleteAllMovies() to skip movies being processed
- Add conflict response (409) in controller for processing movies
- Fixes crash when trying to delete movies during FFmpeg conversion

Commit:
only check conversion status

asebrech committed:
Merge branch 'staging' into delete-video

NotJustJoe commented on apps/back/app/controllers/torrent_controller.ts:
```diff
@@ -40,4 +59,97 @@ export default class TorrentController {
+          success: false,
+          message: 'Invalid movie ID provided',
+        })
+      }
```
Utilises les validators de vinejs pour valider les inputs.
Exemple [ici](https://github.com/asebrech/Hypertube/blob/staging/apps/back/app/validators/comment.ts#L6-L10) avec les commentaires, et tu peux l'utiliser [comme ca](https://github.com/asebrech/Hypertube/blob/staging/apps/back/app/controllers/comments_controller.ts#L114) pour valider les params ensuite.

NotJustJoe commented:
il y'a des conflits a resoudre @asebrech

NotJustJoe committed:
Merge branch 'staging' into delete-video

Commit:
vinejs validation

Commit:
fix validator

asebrech commented:
@NotJustJoe  c'est good

NotJustJoe committed:
fix params error
